lib/cudnn/src/convolution.jl

```diff
@@ -169,6 +169,12 @@ end
 
 ## Utilities to find a fast algorithm
 
+# Helper fct to recover cudnn descriptor tuples from cudnn descriptor pointers
+# so that we can cache algorithms based on data descriptors.
+# Actually just reverses the cache dict and returns the descriptor as a tuple.
+map_cudnn_ptr_to_jl_tuple(cache_dict, desc_ptr) = Dict(zip(values(cache_dict),
+                                                           keys(cache_dict)))[desc_ptr]
```

Comment:
Instead of reversing through and searching the cache every time (expensive!), CUDA.jl provides functions for pulling out the info from a descriptor (cheap!). See https://github.com/JuliaGPU/CUDA.jl/blob/864ec5ee76a623227a73a762b5b6cca048224fd0/lib/cudnn/src/tensor.jl#L49-L72. You'll have to write equivalent functions for some of the conv-specific descriptors, but it should be quite straightforward.

Reply:
Done.

Reply:
The descriptors as they are still have some `Cenum` types in them, which we could convert to julia Ints or something if we run into serialization trouble.

Reply:
Removed.